commit 1ca7862a128e955cd1ff2e7d57daafdbceb1fdd0
Author: Friedrich W. H. Kossebau <kossebau@kde.org>
Date:   Tue Dec 12 17:37:24 2017 +0100

    Require only KF5 components linked to

plus d8a7017ec15955401ef30456f2c9574f7a5736e1, a62d42bab15b0727f1e6b05c636e937cc1be3a8d,
b7328e8a149e26478cef22cff23ad3048f4b9267, 029a04b10b0287500ee06ae08e8775670ce5f8bf

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b3ce3ed1418a505aa990cceb5b1931c176529911..4cc9c6ae153564da3ade3bed2d39db983e0a5da6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,18 +21,18 @@ include(WriteBasicConfigVersionFile)
 include(CheckFunctionExists)
 include(KDEInstallDirs)
 include(KDECMakeSettings)
+include(CheckCXXCompilerFlag)
 
-find_package(KF5 REQUIRED COMPONENTS
-  Config
-  IconThemes
-  I18n
-  ItemModels
-  ItemViews
-  KCMUtils
-  KIO)
+set(KF5_DEP_VERSION "5.15.0")
+find_package(KF5 ${KF5_DEP_VERSION} REQUIRED COMPONENTS
+    I18n
+)
 
 set(KDEVPLATFORM_VERSION "5.1.40")
-find_package(KDevPlatform ${KDEVPLATFORM_VERSION} CONFIG REQUIRED)
+find_package(KDevPlatform ${KDEVPLATFORM_VERSION} CONFIG)
+set_package_properties(KDevPlatform PROPERTIES
+    TYPE REQUIRED
+)
 
 include_directories(
     ${mercurial_BINARY_DIR}
@@ -44,7 +44,6 @@ add_definitions(
     -DQT_DISABLE_DEPRECATED_BEFORE=0x050400
     -DQT_NO_URL_CAST_FROM_STRING
     -DQT_STRICT_ITERATORS
-    -DQT_USE_FAST_CONCATENATION
     -DQT_USE_FAST_OPERATOR_PLUS
 )
 
@@ -62,6 +61,7 @@ if (HAVE_TUC_FLAG)
 endif()
 
 ########### next target ###############
+add_definitions( -DTRANSLATION_DOMAIN=\"kdevmercurial\" )
 
 ecm_qt_declare_logging_category(kdevmercurial_LOG_SRCS
     HEADER debug.h
